astropy/io/ascii/tests/test_read.py

```diff
@@ -1493,3 +1495,43 @@ def test_kwargs_dict_guess(enable):
     for k in get_read_trace():
         if not k.get('status', 'Disabled').startswith('Disabled'):
             assert k.get('kwargs').get('fast_reader').get('enable') is enable
+
+
+@pytest.mark.parametrize('rdb', [False, True])
+@pytest.mark.parametrize('fast_reader', [False, 'force'])
+def test_deduplicate_names_basic(rdb, fast_reader):
+    """Test that duplicate column names are successfully de-duplicated for the
+    basic format.
+    """
+    lines = ['a a_2 a_1 a a']
+    if rdb:
+        lines += ['N N N N N']
+    lines += ['1 2 3 4 5', '10 20 30 40 50']
+
+    if rdb:
+        lines = ['\t'.join(line.split()) for line in lines]
+
+    dat = ascii.read(lines, fast_reader=fast_reader)
+    assert dat.colnames == ['a', 'a_2', 'a_1', 'a_3', 'a_4']
+
+    with ExitStack() as stack:
```

Comment:
This is so complicated, we need a test tp test that the test tests what we think it does. I suggest splitting this in two tests.

1) ``@pytest.mark.parametrize(scenario, [[False, False], [False, 'force'], [True, False])``
2) a test with no parameters for `rdb=True` and `fast_Reader='force'`

Yes, approximate 8 lines of code will be copied, but it might be easier to read.

Disclaimer: It's entirely possible that this suggestion is inappropriate and I just don't understand what the test is doing.